README.md

```diff
@@ -1 +1,1038 @@
+
+[Back to the Top](#top)
+
+To compare environments provided by ForgeRock components, we will create a script that will make an HTTP call to an online service and receive a response in the form of JSON. For this purpose, as an example, we will visit a dummy API employee record at `http://jsonplaceholder.typicode.com/users/1`. This endpoint returns JSON, which the script _could_ evaluate against other data provided by the script's environment.
+
+> If you use server-side scripts to access an API over encrypted connection, make sure Java the script engine is running on trusts the underlying SSL/TLS certificate.
+
+## <a id="chapter-060"></a>[ForgeRock Access Management](https://www.forgerock.com/platform/identity-management) (AM)
+
+[Back to the Top](#top)
+
+AM documentation can be found at [https://backstage.forgerock.com/docs/am](https://backstage.forgerock.com/docs/am).
+
+AM provides authentication and authorization services, and custom scripts can be used to augment the default functionality.
+
+### Managing Scripts in AM
+
+The [Managing Scripts](https://backstage.forgerock.com/docs/am/6.5/authorization-guide/#manage-scripts) chapter in AM's docs shows how the scripts can be managed via REST and command line interfaces. This may be the most efficient way to manage scripts in automatically maintained environments; for example, in production deployments.
+
+In addition, AM console provides a visual and easy to use interface for managing scripts and applying them to authentication and authorization events. We will use the console to apply a script to an authentication procedure in AM.
+
+### Client-Side Scripts
+
+When used for authentication in the front channel, AM allows for creating client-side scripts that would be executed in the user agent.
+
+The use case for a client-side script is collecting information about the user agent's properties and its environment: [Geolocation](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/geolocation), IP, and whatever else that could be collected with a custom script running in a browser. Thus, the script needs to be written in JavaScript compatible with the browser.
+
+From the client side, the data collected with the script can be submitted to the server and become available to the server-side scripts involved in the same authentication procedure.
+
+### Server-Side Scripts
+
+The decision making process on user identification and access management can be aided with the server-side scripts. The server-side scripts can be written in Groovy or JavaScript running in [Rhino](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Rhino).
+
+The server-side scripts can accept data from the client side via a well-know variable.
+
+### Authentication Trees and Authentication Chains
+
+AM supports two basic authentication workflows: `trees` and `chains`. The latter approach allows for use of a client-side script defined directly in AM console.
```

Comment:
You can't have a client-side script in a tree node?

Reply:
(Misplaced the original comment):

I think you can during node development; I saw some on GitHub. But not within AM console, if I remember correctly.

Reply:
So, in the decision node script, one can use the Backchannel ScriptTextOutputCallback in order to run arbitrary JavaScript in user's browser.

https://backstage.forgerock.com/docs/am/6.5/dev-guide/index.html#backchannel-callbacks